src/pyhf/utils.py

```diff
@@ -1,11 +1,25 @@
 import json
 import jsonschema
-import pkg_resources
 from pathlib import Path
 import yaml
 import click
 import hashlib
 
+import sys
+
+if sys.version_info >= (3, 9):
+    from importlib import resources
+else:
+    import importlib_resources as resources
+schemas = resources.files('pyhf') / "schemas"
+
+import pyhf.exceptions
+
+if sys.version_info >= (3, 9):
+    from importlib import resources
+else:
+    import importlib_resources as resources
```

Comment:
Is this double block of 

```python
if sys.version_info >= (3, 9):
    from importlib import resources
else:
    import importlib_resources as resources
```

intentional? If so, why do we need it?

Can we also make it clear that the reason the check for Python 3.9 is needed is because [`importlib.resources.as_file`](https://docs.python.org/3.10/library/importlib.html#importlib.resources.as_file) wasn't added until Python 3.9?

Reply:
Thanks @kratsg for taking up my suggestion!

I second @matthewfeickert's suggestion to change schemas via function call rather than assignment. It is much simpler to add functionality to a function if it ever becomes necessary, than to replace an entire existing module with a class instance with a property.

I'd even go so far to say that a context manager doubling as an update function would be ideal IMO:
```python
# in pyhf.utils
_SCHEMAS = Path(...)
class use_schema_path: # snake_case to remind of function-like usage
    def  __init__(self, path):
        global _SCHEMAS
        self._old_schemas = _SCHEMAS
        _SCHEMAS = pathlib.Path(path)
    def __enter__(self):
        pass
    def __exit__(self, *args, **kwargs):
        global _SCHEMAS
        _SCHEMAS = self._old_schemas
```
which can still be called as a function (only executing `__init__`), so short scripts etc. are not forced to use `with` blocks.
But it can also be used like so:
```python
def make_my_workspace(spec):
    with pyhf.utils.use_schema_path('/my/very/special/schemas'):
        return pyhf.Workspace(spec)
```
So as a user writing code on top of pyhf, I don't have to worry about resesetting the global variable, the CM does it for me, and there are fewer mistakes to make.

Reply:
resolving this, made the changes.